### PR TITLE
feat (help): allow cli option to list tasks with no description

### PR DESCRIPTION
**Pull request by therealkevinard**
added an add'l cli option that lists all tasks, with or without description.
orig. behavior: task -l lists tasks with desc field
new behaviour: task -la or task -a will list all tasks. if task has desc, it will be included.

BREAKING CHANGES: none, that I know of.
NOTES/Concerns:
- This is wip.
- Haven't checked how it interacts with bash completion.
- The new Executor.TaskNames func does not use e.CompiledTask(taskfile.Call{Task: task.Task})

This relates to https://github.com/go-task/task/issues/383

### CODE REVIEW

**Comment by therealkevinard** [comment]
I'm not too emotionally attached to the actual flag, but I went with `-a` (instead of `-L`), because it's more in line with the conventional flag for this sort of thing (eg: `ls -lha`, `du -ah`, and others).

**Comment by kerma** [comment]
Hey @therealkevinard! Would you like to update and finish this?

**Comment by therealkevinard** [comment]
Franky: oh, wow, this pr is stale! 

Yeah, I'd like to knock the dust off it. I'll catch myself up over the next day or so and finish this off. 

Are there any concerns with this note 
> The new Executor.TaskNames func does not use e.CompiledTask(taskfile.Call{Task: task.Task})

I can grok CompiledTask to see what the implications are, but maybe someone here just... Knows? :)

File: cmd/task/task.go [hunk 1]
```diff
@@ -148,8 +150,8 @@ func main() {
 		return
 	}
 
-	if list {
-		e.PrintTasksHelp()
+	if list || listAll {
+		e.PrintTasksHelp(listAll)
 		return
 	}
```
Comment (andreynering) [inline cmd/task/task.go:156]: Boolean arguments are usually discouraged for readability reasons. I'd consider having a separate function for each one:

```suggestion
	if list {
		e.ListTasksWithDesc()
		return
	}
	if listAll {
		e.ListAllTasks()
		return
	}
```

In the package you could then abstract them in s private function as below:

```go
func ListTasksWithDesc() {
 	listTasks(false)
}

func ListAllTasks() {
 	listTasks(true)
}

func listTasks(all bool) {
 	// ...
}
```

File: help.go [hunk 2]
```diff
@@ -26,6 +35,15 @@ func (e *Executor) PrintTasksHelp() {
 	w.Flush()
 }
 
+func (e *Executor) taskNames() (tasks []*taskfile.Task) {
```
Comment (andreynering) [inline help.go:38]: `taskNames` -> `allTaskNames`

File: task.go [hunk 3]
```diff
@@ -64,7 +64,9 @@ func (e *Executor) Run(ctx context.Context, calls ...taskfile.Call) error {
 	for _, c := range calls {
 		if _, ok := e.Taskfile.Tasks[c.Task]; !ok {
 			// FIXME: move to the main package
-			e.PrintTasksHelp()
+			// FIXME: (ard.kevin.84@gmail.com) changed the PrintTasksHelp signature to support show all/some.
+			// 	False preserves original behavior, but should be reviewed.
+			e.PrintTasksHelp(false)
```
Comment (andreynering) [inline task.go:69]: The note is not needed IMO. Let's keep the default behavior which is to list documented tasks only.

```suggestion
			e. ListTasksWithDesc()
```

**Review by andreynering — COMMENTED** [review]
Hi @therealkevinard, thanks for your PR!

I just did a few comments. Let me know of your thoughts.

File: cmd/task/task.go [hunk 4]
```diff
@@ -75,6 +76,7 @@ func main() {
 	pflag.BoolVarP(&helpFlag, "help", "h", false, "shows Task usage")
 	pflag.BoolVarP(&init, "init", "i", false, "creates a new Taskfile.yml in the current folder")
 	pflag.BoolVarP(&list, "list", "l", false, "lists tasks with description of current Taskfile")
+	pflag.BoolVarP(&listAll, "list-all", "a", false, "list tasks with or without a description")
```
Comment (andreynering) [inline cmd/task/task.go:79]: No strong opinion on `-a` vs. `-L`. If anyone has thoughts, let me know.

File: cmd/task/task.go (hunk 1 above)
Comment (therealkevinard) [reply cmd/task/task.go]: resolved in https://github.com/therealkevinard/task/commit/42702e81b35c5b4960c94ff65577f6c4314b0ccd

File: help.go (hunk 2 above)
Comment (therealkevinard) [reply help.go]: resolved in https://github.com/therealkevinard/task/commit/42702e81b35c5b4960c94ff65577f6c4314b0ccd

File: task.go (hunk 3 above)
Comment (therealkevinard) [reply task.go]: resolved in https://github.com/therealkevinard/task/commit/42702e81b35c5b4960c94ff65577f6c4314b0ccd

**Comment by therealkevinard** [comment]
There are no merge conflicts, but this branch obviously behind. Do you think a backmerge is in order?

**Comment by andreynering** [comment]
Hi @therealkevinard, thanks for contributing!

I just merged this and did small changes at 7e0346d6eb04a54e19106611851e4533fe23815c.

**Pull request merged**